clients/githubrepo/client.go

```diff
@@ -228,6 +228,7 @@ func (client *Client) cleanup() error {
 	if err := os.RemoveAll(client.tempDir); err != nil && !os.IsNotExist(err) {
 		return fmt.Errorf("os.Remove: %w", err)
 	}
+	client.files = nil
```

Comment:
How is this causing performance issues? It would be helpful if you can add comments to the code so that in the future it can help others when they are fixing issues.

Reply:
If we don't clean this up, the list of files we iterate through keeps growing. Added comment.

Reply:
I suspect it's because the code keeps appending to the `files`. So the code ends up iterating over all previous's project files, instead of just the one it should be looking at.

Reply:
Thanks! @azeemshaikh38